### DISCUSSION TITLE
Updating Client Model View

### DISCUSSION THREAD

**Discussion by JValGLZ** [eclipse-glsp/glsp#1325] [Q&A]
I have created a property panel that can be used to change attributes in nodes. The issue I am having is that after I apply the changes, it does not reflect on the client node that was changed. The change does reflect whenever I drag the node. What am I missing so that it updates immediately after pressing the apply button? The apply button dispatches an operation. Thank you!

**Comment by JValGLZ** [comment]
I have found that it is by dispatching the UpdateModelAction from the server side. I have encountered another issue though. When I press apply, the labels inside the node go to the top left corner of the node. After moving it, the issue is fixed. Why does this occur? 


https://github.com/eclipse-glsp/glsp/assets/108287980/aaf84749-7e63-41a8-bd2a-f978fe9c4f9c

**Comment by tortmayr** [comment]
Hi @JValGLZ,
>I have found that it is by dispatching the UpdateModelAction from the server side. I have encountered another issue though. When I press apply, the labels inside the node go to the top left corner of the node. After moving it, the issue is fixed. Why does this occur?

Manually dispatching an `UpdateModelAction` from the server side should not be necessary. This indicates that there is an underlying problem in our implementation. If your apply operation is properly handled by an operation handler on the server side, the server automatically takes care of updating the underlying model. In general, the update cycle includes more than just dispatching a `UpdateModelAction`. Typically the update cycle looks as follows: 
1. Server handles operation -> dispatches `RequestBoundsAction` 
2. Client handles `RequestBoundsAction` and computes the new bounds for the updated model and dispatches a  `ComputedBoundsActions`
3. Server handles `ComputedBoundsAction`, applies it to the model and sends an `UpdateModelAction`
4. Client handles `UpdateModelAction` and re renders the model.

If you operation is correctly handled by an `OperationHandler` the whole update process is invoked automatically and there should be no need for a manual `UpdateModelAction`.

Could you please share your implementation of the apply operation handler on the server side?
